### PR TITLE
Add support for collecting label statistics for classification tasks during graph construction.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
#264 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by zhjwy9343 — APPROVED** [review]
LGTM, except for a few naming conventions.

File: python/graphstorm/gconstruct/README.md [hunk 1]
```diff
@@ -75,6 +75,13 @@ file names that contains training/validation/test node IDs. The keys
 of the dict are "train", "valid" and "test". A node ID in the files
 is stored as a JSON object. Custom data split currently only works for
 nodes.
+* `label_stats_type` specifies how to collect label statistics of the training set.
+This is optional. If it's not specified, no label statistics are collected.
+If specified, the label statistics will be saved as `node_label_stats.json` or
+`edge_label_stats.json` under the output directory for node tasks or edge tasks,
```
[the remark this answers is not in the document]
Comment (classicsong) [reply python/graphstorm/gconstruct/README.md:81]: Yes.
We will create a json file.

[the remark this answers is not in the document]
Comment (classicsong) [reply python/graphstorm/gconstruct/README.md:81]: Updated.

File: python/graphstorm/gconstruct/transform.py [hunk 2]
```diff
@@ -31,6 +32,151 @@
 
 from .file_io import HDF5Array, read_index_json
 
+LABEL_STATS_FIELD = "training_label_stats"
```
[the remark this answers is not in the document]
Comment (classicsong) [reply python/graphstorm/gconstruct/transform.py:35]: I would prefer training_label_stats as we are collecting stats of train labels.

**Pull request merged**